edvart/utils.py

```diff
@@ -126,23 +127,60 @@ def reindex_to_period(
     return df
 
 
-def discrete_colorscale(n, saturation=0.5, lightness=0.5) -> Iterable[Tuple[float, str]]:
+def hsl_wheel_colorscale(n: int, saturation=0.5, lightness=0.5) -> Iterable[str]:
     """
     Generate a colorscale of n discrete colors.
 
-    Colours are equally spaced around the complete HSL wheel with constant saturation and lightness.
+    Colors are equally spaced around the complete HSL wheel with constant saturation and lightness.
 
     Returns
     -------
     Iterable[Tuple[float, str]]
-        An iterable of 2n tuples, where each tuple contains a value between 0 and 1
-        (the values are equally and each value appears twice) and an HSL string containing
-        an HSL color string with hue corresponding to the value.
+        An iterable of n plotly-compatible HSL strings.
     """
     for i in range(n):
-        color = f"hsl({(i / n) * 360 :.2f}, {saturation * 100 :.2f}%, {lightness * 100 :.2f}%)"
-        yield (i / n, color)
-        yield ((i + 1) / n, color)
+        yield f"hsl({(i / n) * 360 :.2f}, {saturation * 100 :.2f}%, {lightness * 100 :.2f}%)"
+
+
+def make_discrete_colorscale(colorscale: List[str], n_colors: int) -> Iterable[Tuple[float, str]]:
+    """
+    Generate a colorscale of n discrete colors.
+
+    Returns
+    -------
+    Iterable[Tuple[float, str]]
+        An iterable of 2n tuples, where each tuple contains a value between 0 and 1
+        (the values are equally and each value appears twice) and one of the colors from the
+        `colorscale`.
+    """
+    for i in range(n_colors):
+        color = colorscale[i % len(colorscale)]
+        yield (i / n_colors, color)
+        yield ((i + 1) / n_colors, color)
```

Comment:
It took me a while to understand that this "double yield" is to make a the colorscale discrete. I would either:
 1. Make a comment here explaining that (each color appears twice in the continous colorscale while the second has the same "coordinate", the float number, as the next color which makes the colorscale discrete.
 2. Use `color_discrete_sequence` https://plotly.com/python/discrete-color/.

If option 2 cannot be used (e.g. used charts do not have the arg) I'd add a comment explaining why it cannot be used.

Reply:
`color_discrete_sequence` is only available in `plotly.express` (a higher-level API for plotly). The plots where this colorscale is used are done with `plotly.graph_objects`, which doesn't have that parameter. The plots could probably be rewritten in `plotly.express`, but it would be non-trivial.

Added a comment noting that.